Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp

```diff
@@ -126,16 +126,30 @@ UiProfile Spire::make_icon_button_profile() {
   populate_widget_properties(properties);
   properties.push_back(make_standard_qstring_property("tooltip",
     QString::fromUtf8("Tooltip")));
+  properties.push_back(make_standard_bool_property("checkable"));
+  properties.push_back(make_standard_bool_property("checked"));
   auto profile = UiProfile(QString::fromUtf8("IconButton"), properties,
     [] (auto& profile) {
       auto button = new IconButton(imageFromSvg(":/Icons/demo.svg",
         scale(26, 26)));
+      apply_widget_properties(button, profile.get_properties());
       auto& tooltip = get<QString>("tooltip", profile.get_properties());
       button->setToolTip(tooltip.get());
-      apply_widget_properties(button, profile.get_properties());
       tooltip.connect_changed_signal([=] (const auto& value) {
         button->setToolTip(value);
       });
+      auto& checkable = get<bool>("checkable", profile.get_properties());
+      checkable.connect_changed_signal([=] (auto is_checkable) {
+        button->setCheckable(is_checkable);
+        button->update();
```

Comment:
When the checkable is set to false, the checked which is true should be reset to false.

Reply:
@kranar recommended in the case notes that I don't add too much complexity to the tester: "You can program any arbitrary logic you want, but I would recommend against it."

It's a small inconsistency with the tester, but it's not caused by an issue with the button itself.

Reply:
I see.